### ISSUE TITLE
No diagnostic rule for overridden final methods

### ISSUE THREAD

**Issue by hamdanal** [microsoft/pyright#6985]
**Describe the bug**

If a `@final` decorated method is overridden, pyright emits an error with no diagnostic rule. This forces me to use a blanket `# pyright: ignore` to silence the error.

**Code or Screenshots**
```python
from typing import final

class Base:
    @final
    def f(self) -> None: ...

class Child(Base):
    @final
    def f(self) -> None: ...
```

or via [the playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMrCoCGANgFBkDGJRAzrVAEJ0CmAXGVF1AAKEqlO3ACYtgBABS0WJYAEooAWgB8UAHJgU7KADo9FanQYBhABZISwic2lyO3Xv0EPR44FJnylqjVra79IA)

**VS Code extension or command-line**

command-line: pyright 1.1.345

**Comment by erictraut** [comment]
Thanks for the bug report. Yes, this should be covered by the `reportIncompatibleMethodOverride` rule.

This will be fixed in the next release.

**Comment by hamdanal** [comment]
You should win "The fastest maintainer award" :D

Thank you for the quick fix.

**Comment by erictraut** [comment]
This is included in pyright 1.1.346, which I just published. It will also be included in a future release of pylance.
